test/ng/directive/inputSpec.js

```diff
@@ -2465,6 +2465,21 @@ describe('input', function() {
         expect($rootScope.form.alias.$error.min).toBeFalsy();
       });
 
+
+      it('should validate against the viewValue', function() {
+        var inputElm = helper.compileInput('<input type="number" num-parse ng-model="value" name="alias" min="10" />');
+        var ngModelCtrl = inputElm.controller('ngModel');
+        ngModelCtrl.$parsers.push(function(value) {
+          return value - 5;
+        });
+
+        helper.changeInputValueTo('10');
+        expect(inputElm).toBeValid();
+        expect($rootScope.value).toBe(5);
+        expect($rootScope.form.alias.$error.min).toBeFalsy();
```

Comment:
Should we also have tests that check that we *do* get an error if the `viewValue` is bad but the `modelValue` is good?